phabricator.js

```diff
@@ -0,0 +1,157 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+let parentRevision = null;
+let failedToFetch = false;
+
+async function injectButtons() {
```

Comment:
Let's split this function into multiple functions, otherwise it gets hard to read

Reply:
Done. I still put all the file-related functions inside `injectButton()` to use the bindings.

Reply:
You could also pass arguments to the functions. It's fine by me either way.